src/Microsoft.Azure.WebJobs.ServiceBus/MessageProcessor.cs

```diff
@@ -58,20 +70,22 @@ public MessageProcessor(MessageHandlerOptions messageOptions)
         /// <returns>A <see cref="Task"/> that will complete the message processing.</returns>
         public virtual async Task CompleteProcessingMessageAsync(Message message, FunctionResult result, CancellationToken cancellationToken)
```

Comment:
In reference to my previous comment -- it may make more sense to pass the `MessageReceiver` into this method now to ensure we're using the same one that received the message.

Reply:
I added  MessageReceiver cache like we have for MessagingFactory in v1:
https://github.com/Azure/azure-webjobs-sdk/blob/e915ee574136207cab3b01cac7139bf09b013ace/src/Microsoft.Azure.WebJobs.ServiceBus/MessagingProvider.cs#L24